src/Common/tests/System/AssertExtensions.cs

```diff
@@ -169,5 +193,31 @@ private static string AddOptionalUserMessage(string message, string userMessage)
             if (actual.CompareTo(lessThanOrEqualTo) > 0)
                 throw new XunitException(AddOptionalUserMessage($"Expected: {actual} to be less than or equal to {lessThanOrEqualTo}", userMessage));
         }
+
+        /// <summary>
+        /// Validate that a given value is greater than or equal to another value.
+        /// </summary>
+        /// <param name="actual">The value that should be greater than or equal to <paramref name="greaterThanOrEqualTo"/></param>
+        /// <param name="greaterThanOrEqualTo">The value that <paramref name="actual"/> should be greater than or equal to.</param>
+        public static void GreaterThanOrEqualTo<T>(T actual, T greaterThanOrEqualTo, string userMessage = null) where T : IComparable
```

Comment:
It is a little bit more than that:

``` c#
            if ((actual == null && greaterThanOrEqualTo != null) || (actual != null && actual.CompareTo(greaterThanOrEqualTo) < 0))
                throw new XunitException(AddOptionalUserMessage($@"Expected: {actual?.ToString() ?? Null} to be greater than or equal to {greaterThanOrEqualTo?.ToString() ?? Null}", userMessage));
```

I opted for clearer and more verbose in these cases as this is a test assertion. I'm ok with merging the other one (`LessThanOrEqualTo`), but I'd rather keep the layout the same to make visually comparing the two methods easier.